lib/workflow/backend/backend.go

```diff
@@ -0,0 +1,137 @@
+package backend
```

Comment:
We think we want to comment these types a bit as internal documentation.  We'll likely want to do the same for the public API.